UM/Qt/Bindings/ProfilesModel.py

```diff
@@ -192,6 +211,31 @@ def getFileNameFiltersWrite(self):
         filters.append(catalog.i18nc("@item:inlistbox", "All Files (*)")) #Also allow arbitrary files, if the user so prefers.
         return filters
 
+    def _onMachineInstanceChanged(self):
+        if self._working_profile:
+            self._working_profile.settingValueChanged.disconnect(self._onWorkingProfileValueChanged)
+        self._working_profile = self._manager.getWorkingProfile()
+        if self._working_profile:
+            self._working_profile.settingValueChanged.connect(self._onWorkingProfileValueChanged)
+
+        self._onProfilesChanged()
+
+        #Restore active profile for this machine_instance.
+        active_instance_name = self._manager.getActiveMachineInstance().getActiveProfileName()
+        active_profile = self._manager.findProfile(active_instance_name, instance = self._manager.getActiveMachineInstance())
+
+        if not active_profile:
+            #A profile by this name is no longer in the filtered list of profiles.
+            profiles = self._manager.getProfiles(instance = self._manager.getActiveMachineInstance())
+            for profile in profiles:
+                active_profile = profile #Default to first profile you can find.
+                break
```

Comment:
Could it happen that a machine instance has no profiles at all? Or do we require that for every filter there is at least one profile?